scripts/build/utils.mjs

```diff
@@ -0,0 +1,35 @@
+import fs from "node:fs/promises";
+import url from "node:url";
+import path from "node:path";
+import { createRequire } from "node:module";
+
+async function readJson(file) {
+  const data = await fs.readFile(file);
+  return JSON.parse(data);
+}
+
+function writeJson(file, content) {
+  content = JSON.stringify(content, null, 2);
+  return fs.writeFile(file, content);
+}
+
+async function copyFile(from, to) {
+  const data = await fs.readFile(from);
+  return fs.writeFile(to, data);
+}
+
+function commonjsObjects(importMeta) {
+  return {
+    get __filename() {
+      return url.fileURLToPath(importMeta.url);
+    },
+    get __dirname() {
+      return path.dirname(url.fileURLToPath(importMeta.url));
+    },
+    get require() {
+      return createRequire(importMeta.url);
+    },
+  };
+}
```

Comment:
This is super useful utils for me, so I published the package [`esm-migration-utils`](https://github.com/sosukesuzuki/esm-migration-utils). Probably we will use this several times in the future in our codes. Can we use the package?

Are you sure you want to list your name in the README? like `This package is extracted from fisker's code.`Also, I can remove this package if you don't like it.

Reply:
I have `esm-utils` module, haha.

Reply:
Oh, I think we can use it on Prettier.

Reply:
> Are you sure you want to list your name in the README? like This package is extracted from fisker's code.

Not necessary

> Also, I can remove this package if you don't like it.

Feel free to do whatever with it.

Reply:
> I think we can use it on Prettier.

I don't like promoting my own packages, but I just published `v1.0.0`, I'll use it in this PR.